src/charm.py

```diff
@@ -116,19 +122,35 @@ def _get_database_relation_info(self) -> dict:
         relation_data = self.database.fetch_relation_data()[relation_id]
 
         return {
-            "username": relation_data["username"],
-            "password": relation_data["password"],
-            "endpoints": relation_data["endpoints"],
+            "username": relation_data.get("username"),
+            "password": relation_data.get("password"),
+            "endpoints": relation_data.get("endpoints"),
         }
 
-    def _update_layer(self) -> None:
-        """Updates the Pebble configuration layer and config if changed."""
-        self.unit.status = MaintenanceStatus("Applying pebble layer")
+    def _run_sql_migration(self, set_timeout) -> None:
+        """Runs a command to create SQL schemas and apply migration plans."""
+        process = self._container.exec(
+            ["hydra", "migrate", "sql", "-e", "--config", self._hydra_config_path, "--yes"],
+            timeout=20.0 if set_timeout else None,
+        )
 
-        if not self._container.get_plan().to_dict():
-            # Push the config on first layer update to avoid PathError
-            self._container.push(self._hydra_config_path, self._config, make_dirs=True)
-            logger.info("Pushed hydra config")
+        stdout, _ = process.wait_output()
+        logger.info(f"Executing automigration: {stdout}")
+
+    def _on_hydra_pebble_ready(self, event) -> None:
+        """Event Handler for pebble ready event."""
+        if not self.model.relations["pg-database"]:
+            logger.error("Missing required relation with postgresql")
+            self.model.unit.status = BlockedStatus("Missing required relation with postgresql")
+            return
+
```

Comment:
Here should we have a condition that validates if the database is created and if not is placed in waiting state (`if not self.database.is_database_created`). This would mean that we 
1. Check if relation is created -> else block
2. Check if database is created -> else wait
3. Check if can connect to workload -> else wait

Then the rest (configuring pebble) is done without if/elses

Reply:
I am fine with that approach, but we agreed to follow a pattern in kratos and hydra to add the layer before checking if the database is created.
Is it fine to:
1. Check if relation is created -> else block
2. Check if can connect to workload -> else wait
3. _Add layer and replan_
4. Check if database is created -> else wait?

Reply:
My concern is with the order of 3/4. It doesn't make much sense to me to create the layer before the database is created. Following this logic, why check if the relation with the db is created before adding the layer?

Reply:
I reordered it to check both the relation and database presence after the layer is added. I think it's fine to create it even before the database is created, as we only need to push a config file then.